src/targets.rs

```diff
@@ -541,15 +525,15 @@ impl Target {
         }
     }
 
-    pub fn create_target_machine(&self, triple: &str, cpu: &str, features: &str, level: Option<CodeGenOptLevel>, reloc_mode: RelocMode, code_model: CodeModel) -> Option<TargetMachine> {
+    pub fn create_target_machine(&self, triple: &str, cpu: &str, features: &str, level: Option<OptimizationLevel>, reloc_mode: RelocMode, code_model: CodeModel) -> Option<TargetMachine> {
```

Comment:
We should probably use `OptimizationLevel` with None value or `Option<OptimizationLevel>`, but not both?

Reply:
Well, `None` is a member of  `OptimizationLevel` in LLVM, so I'd like to keep it in the enum; and I think `Option<OptimizationLevel>` means something like "either choose your own opt level, or we'll choose the default one for you", but it does seem redundant. Want me to change this to `OptimizationLevel`, and maybe impl `Default` for it?

Reply:
It's times like this that make me wish rust had optional arguments haha. Yeah I guess `OptimizationLevel` with a default impl is fine

Reply:
Alright, I'll do it this way. Btw, I sent you some messages on the Gitter.